### PR TITLE
docs: Reformat user guide into feature sections

### PR DESCRIPTION
**Pull request by corycaywood**
# Description
Sorry it looks like there are a lot of changes. Most of the content is still the same but was just rearranged. The only new content is the Mini App Features section which explains what each feature does. Also added a Table of Contents - this will be automatically generated by Jekyll when the documentation is published to Github Pages. Also put some of the longer code examples inside of expandable sections.

It's probably easier to preview it directly on my branch: 
https://github.com/corycaywood/android-miniapp/blob/docs/userguide-reformat/miniapp/USERGUIDE.md

## Links
MINI-1978

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [ x I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,7 +63,22 @@ $ ./gradlew assembleRelease
 
 We currently don't provide an API for public use, so you must provide your own API.
 
-## How to generate KDocs SDK documentation locally
+## Writing and generating documentation
+
+Our documentation is hosted on Github Pages using the `gh-pages` branch of this repo. So this means that the docs are hosted as markdown and then Github Pages generates HTML using Jekyll. The documentation has two parts: a userguide and the API docs. The userguide is generated from [USERGUIDE.md](miniapp/USERGUIDE.md) and the API docs are generated using Dokka.
+
+For the most part, you cn use standard markdown in the userguide, but please note the following:
```
Comment (khairul-alam-licon) [inline README.md:70]: cn ➞ can 🤔

File: README.md [hunk 2]
```diff
@@ -76,7 +91,7 @@ The docs are hosted on Github Pages in markdown, and therefore the HTML version
 ```
 $ ./gradlew generatePublishableDocs
 $ git checkout gh-pages
-$ cp miniapp/build/publishableDocs/docs ./
+$ cp -r miniapp/build/publishableDocs/docs/ ./
```
Comment (khairul-alam-licon) [inline README.md:94]: 👍

File: miniapp/USERGUIDE.md [hunk 3]
```diff
@@ -444,6 +580,23 @@ CoroutineScope(Dispatchers.Main).launch {
     // This runs on the main thread
 }
 ```
+</details>
+
+<details><summary markdown="span"><b>How do I use snapshot versions of this SDK?</b>
+</summary>
+
+We may peridically publish snapshot versions for testing pre-release features. These versions will always end in `-SNAPSHOT`, for example `1.0.0-SNAPSHOT`. If you wish to use a snapshot version, you will need to add the snapshot repo to your Gradle configuration.
```
Comment (khairul-alam-licon) [inline miniapp/USERGUIDE.md:588]: peridically ➞ periodically

File: miniapp/USERGUIDE.md [hunk 4]
```diff
@@ -32,21 +38,22 @@ dependency {
...
 | RAS Project Subscription Key | String  | `com.rakuten.tech.mobile.ras.ProjectSubscriptionKey`   | ❌         | 🚫        |
 | Host App User Agent Info     | String  | `com.rakuten.tech.mobile.miniapp.HostAppUserAgentInfo` | ✅         | 🚫        |
 
 **Note:**  
 * We don't currently host a public API, so you will need to provide your own Base URL for API requests.
-* The host app info is the string which is appended to user-agent of webview. It should be a meaningful keyword such as host app name to differentiate other host apps.
+* The "Host App User Agent Info" is the string which is appended to user-agent of webview. It should be a meaningful keyword such as host app name to differentiate other host apps.
 
-In your `AndroidManifest.xml`:
+<details><summary markdown="span"><b>Click to expand example AndroidManifest.xml</b>
```
Comment (khairul-alam-licon) [inline miniapp/USERGUIDE.md:55]: 🥇

**Review by khairul-alam-licon — APPROVED** [review]
Looks nice 👍

File: README.md (hunk 1 above)
Comment (corycaywood) [reply README.md]: Thanks, I fixed this typo.

File: miniapp/USERGUIDE.md (hunk 3 above)
Comment (corycaywood) [reply miniapp/USERGUIDE.md]: Thanks, I fixed this too.

File: miniapp/USERGUIDE.md [hunk 5]
```diff
@@ -206,52 +291,138 @@ val userInfoBridgeDispatcher = object : UserInfoBridgeDispatcher() {
...
+CoroutineScope(Dispatchers.IO).launch {
+    try {
+        val miniAppInfo = MiniApp.instance().fetchInfo("MINI_APP_ID")
+    } catch(e: MiniAppSdkException) {
+        Log.e("MiniApp", "There was an error retrieving the Mini App info", e)
+    }
+}
+```
+
+**Note:** Preview Mode:
```
Comment (minh-rakuten) [inline miniapp/USERGUIDE.md:419]: I think the note of preview mode should be put under #3 Create and display a Mini App

Comment (corycaywood) [reply miniapp/USERGUIDE.md]: @minh-rakuten Thanks, I've updated this. I also noticed that there were a lot of notes under that section so I re-formatted it into a list of notes.

**Review by minh-rakuten — APPROVED** [review]
I really love the new format.

**Pull request merged**